tools/evm/JsonTypes/EnvInfo.cs

```diff
@@ -0,0 +1,126 @@
+        public UInt256? ParentBaseFee { get; set; }
+        public long ParentGasUsed { get; set; }
+        public long ParentGasLimit { get; set; }
+        public ulong? ParentExcessBlobGas { get; set; }
+        public ulong? CurrentExcessBlobGas { get; set; }
+        public ulong? ParentBlobGasUsed { get; set; }
+        public Dictionary<string, Hash256> BlockHashes { get; set; } = [];
+        public Ommer[] Ommers { get; set; } = [];
+        
+        public bool IsStateTest { get; set; }
+
+        public Hash256? GetCurrentRandomHash256()
+        {
+            if (CurrentRandom == null) return null;
+
+            if (CurrentRandom.Length < Hash256.Size)
+            {
+                var currentRandomWithLeadingZeros = new byte[Hash256.Size];
+                Array.Copy(CurrentRandom, 0, currentRandomWithLeadingZeros, Hash256.Size - CurrentRandom.Length,
+                    CurrentRandom.Length);
+                CurrentRandom = currentRandomWithLeadingZeros;
+            }
+
+            return new Hash256(CurrentRandom);
+        }
+
+        public void ApplyChecks(ISpecProvider specProvider, IReleaseSpec spec)
+        {
+            ApplyLondonChecks(spec);
+            ApplyShanghaiChecks(spec);
+            ApplyCancunChecks(spec);
+            ApplyMergeChecks(specProvider);
+        }
+
+        private void ApplyLondonChecks(IReleaseSpec spec)
+        {
+            if (spec is not London) return;
+            if (CurrentBaseFee != null) return;
+
+            if (!ParentBaseFee.HasValue || CurrentNumber == 0)
+            {
+                throw new T8NException("EIP-1559 config but missing 'currentBaseFee' in env section", ExitCodes.ErrorConfig);
+            }
+        }
+
+        private void ApplyShanghaiChecks(IReleaseSpec spec)
+        {
+            if (spec is not Shanghai) return;
+            if (Withdrawals == null)
+            {
+                throw new T8NException("Shanghai config but missing 'withdrawals' in env section", ExitCodes.ErrorConfig);
+            }
+        }
+
+        private void ApplyCancunChecks(IReleaseSpec spec)
+        {
+            if (spec is not Cancun)
+            {
+                ParentBeaconBlockRoot = null;
+                return;
+            }
+
+            if (ParentBeaconBlockRoot == null)
+            {
+                throw new T8NException("post-cancun env requires parentBeaconBlockRoot to be set", ExitCodes.ErrorConfig);
+            }
+        }
+
+        private void ApplyMergeChecks(ISpecProvider specProvider)
+        {
+            if (specProvider.TerminalTotalDifficulty?.IsZero ?? false)
+            {
+                if (CurrentRandom == null) throw new T8NException("post-merge requires currentRandom to be defined in env", ExitCodes.ErrorConfig);
+                if (CurrentDifficulty?.IsZero ?? false) throw new T8NException("post-merge difficulty must be zero (or omitted) in env", ExitCodes.ErrorConfig);
```

Comment:
The following `env.json` (with a `currentlDifficulty` value of `"0"`) and `--state.fork Cancun` unexpectedly triggers this exception.

```json
{
    "currentCoinbase": "0x2adc25665018aa1fe0e6bc666dac8fc2697ff9ba",
    "currentGasLimit": "100000000000000000",
    "currentNumber": "1",
    "currentTimestamp": "1000",
    "currentRandom": "0",
    "currentDifficulty": "0",
    "currentBaseFee": "7",
    "currentExcessBlobGas": "0",
    "currentBlobGasUsed": "0",
    "parentUncleHash": "0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347",
    "parentBeaconBlockRoot": "0x0000000000000000000000000000000000000000000000000000000000000000",
    "blockHashes": {},
    "ommers": [],
    "withdrawals": []
}
```

I think this resolves it?
```suggestion
                if (CurrentDifficulty != null && !CurrentDifficulty.Value.IsZero) throw new T8NException("post-merge difficulty must be zero (or omitted) in env", ExitCodes.ErrorConfig);
```